jquery.flot.navigate.js

```diff
@@ -117,15 +120,24 @@ can set the default in the options.
             }
         }
 
-        var SNAPPING_CONSTANT = $.plot.uiConstants.SNAPPING_CONSTANT;
-        var PANHINT_LENGTH_CONSTANT = $.plot.uiConstants.PANHINT_LENGTH_CONSTANT;
-
         function onMouseWheel(e, delta) {
             e.preventDefault();
             onZoomClick(e, delta < 0);
             return false;
         }
 
+        function onMouseDown(e) {
```

Comment:
This is a hack that I don't expect to port over into the master branch. Without bothering with tracking the mouse down and up events to determine if we are in a valid possible drag state, everything seems to work in all of the needed browsers. The only thing that was failing was the test under karma.

Ultimately, the odd behavior seems to be the fact that 'OnDragStart' is called at all after releasing the mouse when you finish dragging the cursor. 'OnDragStart', I would assume, should never be called when the left mouse button isn't pressed. Tracking this down will take time, I think, so this seems like an okay-ish intermediate solution.

Reply:
It is worth doing a bit of testing of when mousedown and mouseup are triggered, especially with weirder interactions like starting the click inside the graph and ending it outside the graph. We may find that this hack allows the user to get into some weird states. We may still be OK with those states because they are corner cases, but we should characterize it in case we're actually breaking a valid case.

Reply:
Agreed. Will do.

Reply:
From my own testing, I don't see any degradation in the current behavior with my change. Note that the current behavior isn't ideal when you drag a cursor to the edge of the plot area and continue dragging outside of it. I would think the preferable behavior would be to either allow the cursor to continue moving along with the cursor "off screen", or begin panning the range such that it follows the cursor. CurrentGen does sort of a combination of this, while the desktop NXG just allows the cursor to be dragged "off screen". Still, this is an aside that doesn't need addressing with this change.